Marlin/Marlin_main.cpp

```diff
@@ -2657,42 +2655,50 @@ inline void gcode_G28() {
           SERIAL_PROTOCOLPGM(" d: ");
           SERIAL_PROTOCOL_F(plane_equation_coefficients[2], 8);
           SERIAL_EOL;
-          if (verbose_level > 2) {
-            SERIAL_PROTOCOLPGM("Mean of sampled points: ");
-            SERIAL_PROTOCOL_F(mean, 8);
-            SERIAL_EOL;
-          }
         }
 
         if (!dryrun) set_bed_level_equation_lsq(plane_equation_coefficients);
         free(plane_equation_coefficients);
+        matrix_3x3 inverse_bed_level_matrix = matrix_3x3::transpose(plan_bed_level_matrix); // inverse bed level matrix
+
+        // search minimum and maximum point on bed in rotated coordinats
+        float rot_diff = 0,
+              rot_min_diff = Z_MAX_POS,
+              rot_max_diff = -Z_MAX_POS;
+        for (int ProbeCount = 0; ProbeCount < abl2; ProbeCount++)
+          {
+            vector_3 probe_point = vector_3(eqnAMatrix[ProbeCount + 0 * abl2], eqnAMatrix[ProbeCount + 1 * abl2], eqnBVector[ProbeCount]);
+            probe_point.apply_rotation(inverse_bed_level_matrix);
+            float rot_diff =  probe_point.z ;
+            if (rot_diff < rot_min_diff ) rot_min_diff = rot_diff;
+            if (rot_diff > rot_max_diff ) rot_max_diff = rot_diff;
+          }
+        //SERIAL_PROTOCOLPGM("rot_min_diff=");SERIAL_PROTOCOL_F(rot_min_diff, 5);SERIAL_EOL;
+        //SERIAL_PROTOCOLPGM("rot_max_diff=");SERIAL_PROTOCOL_F(rot_max_diff, 5);SERIAL_EOL;
+        //SERIAL_PROTOCOLPGM("difference=");SERIAL_PROTOCOL_F(rot_max_diff-rot_min_diff, 5);SERIAL_EOL;
 
         // Show the Topography map if enabled
         if (do_topography_map) {
+          // search minimum measured Z
+          float diff = 0,
+                min_diff = Z_MAX_POS;
+          for (int ProbeCount = 0; ProbeCount < abl2; ProbeCount++)
+            {
+              diff = eqnBVector[ProbeCount];
+              if (diff < min_diff ) min_diff = diff;
+            }
```

Comment:
See http://marlinfirmware.org/index.php/Coding_Standards#Bracket-style for the indentation rules. The code here should be spaced like so:
```cpp
          for (int ProbeCount = 0; ProbeCount < abl2; ProbeCount++) {
            diff = eqnBVector[ProbeCount];
            if (diff < min_diff ) min_diff = diff;
          }
```